src/bls12_381/ec.rs

```diff
@@ -10,6 +10,15 @@ macro_rules! curve_impl {
         $compressed:ident,
         $pairing:ident
     ) => {
+
+        fn y2_from_x(x: $basefield) -> $basefield {
```

Comment:
I don't know if this change still needs to be in this PR anymore, it could be separate. I don't think y2_from_x is used directly in sw_encode etc. anymore. But, it's not a bad refactoring.